spec/chewy/fields/base_spec.rb

```diff
@@ -402,33 +402,234 @@
+            City.create!(id: 2, name: '', surname: '', description: '')
+          ]
+
+          Country.create!(id: 1, cities: cities)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq(
+            'id' => 1, 'cities' => [
+              {'id' => 1, 'name' => '', 'surname' => ''},
+              {'id' => 2, 'name' => '', 'surname' => ''}
+            ]
+          )
+        end
       end
 
-      let(:country_with_cities) do
-        cities = [City.create!(id: 1, name: 'City1'), City.create!(id: 2, name: 'City2')]
+      context 'fields with ignore_blank: true flag for nil value and with .empty? method' do
+        before do
+          stub_index(:countries) do
+            define_type Country do
+              field :id
+              field :cities do
+                field :id
+                field :name, ignore_blank: true
+                field :surname, ignore_blank: true
+                field :description, ignore_blank: true
+              end
+            end
+          end
+        end
+
+        let(:country_with_cities) do
+          cities = [
+            City.create!(id: 1, name: '', surname: '', description: nil)
+          ]
 
-        Country.create!(id: 1, cities: cities)
+          Country.create!(id: 1, cities: cities)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq(
+            'id' => 1, 'cities' => [
+              {'id' => 1, 'description' => nil}
+            ]
+          )
+        end
       end
 
-      specify do
-        expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq('id' => 1, 'cities' => [
-          {'id' => 1, 'name' => 'City1'}, {'id' => 2, 'name' => 'City2'}
-        ])
+      context 'parental and children fields with ignore_blank flag' do
+        before do
+          stub_index(:countries) do
+            define_type Country do
+              field :id
+              field :cities, ignore_blank: true do
+                field :id
+                field :name
+                field :surname, ignore_blank: true
+                field :description
+              end
+            end
+          end
+        end
+
+        let(:country_without_cities) do
+          cities = []
+
+          Country.create!(id: 1, cities: cities)
+        end
+
+        let(:country_with_cities) do
+          cities = [
+            City.create!(id: 1, name: '', surname: '', description: nil)
+          ]
+
+          Country.create!(id: 1, cities: cities)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_without_cities))
+            .to eq('id' => 1)
+        end
+
+        specify do
+          expect(CountriesIndex::Country.root.compose(country_with_cities)).to eq(
+            'id' => 1, 'cities' => [
+              {'id' => 1, 'name' => '', 'description' => nil}
+            ]
+          )
+        end
```

Comment:
I would replace this with something like:

```suggestion
        let(:country) { Country.create!(id: 1, cities: cities) }

        context('without cities') do
          let(:cities) { [] }

          specify do
            expect(CountriesIndex::Country.root.compose(country))
              .to eq('id' => 1)
          end
        end

        context('with cities') do
          let(:cities) { [City.create!(id: 1, name: '', surname: '') ] }

          specify do
            expect(CountriesIndex::Country.root.compose(country)).to eq(
              'id' => 1, 'cities' => [
                {'id' => 1, 'name' => '', 'description' => nil}
              ]
            )
          end
        end
```